crates/autopilot/src/driver_model.rs

```diff
@@ -0,0 +1,124 @@
+//! Types for communicating with drivers.
+
+pub mod quote {
+    use model::u256_decimal;
+    use primitive_types::{H160, U256};
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Request {
+        pub sell_token: H160,
+        pub buy_token: H160,
+        pub kind: Kind,
+        #[serde(with = "u256_decimal")]
+        pub amount: U256,
+    }
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    #[serde(rename_all = "lowercase")]
+    pub enum Kind {
+        #[default]
+        Buy,
+        Sell,
+    }
+
+    #[derive(Clone, Debug, Deserialize, Serialize)]
+    #[serde(untagged, rename_all = "camelCase")]
+    pub enum Response {
+        Successful {
+            #[serde(with = "u256_decimal")]
+            sell_amount: U256,
+            #[serde(with = "u256_decimal")]
+            buy_amount: U256,
+            gas: u64,
+        },
+        Unfillable {
+            unfillable_reason: String,
+        },
+    }
+}
+
+pub mod solve {
+    use bigdecimal::num_bigint::BigUint;
+    use chrono::{DateTime, Utc};
+    use primitive_types::H160;
+    use serde::{Deserialize, Serialize};
+    use serde_with::{serde_as, DisplayFromStr};
+    use std::collections::BTreeMap;
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct Request {
+        pub auction: Auction,
+        pub deadline: DateTime<Utc>,
+    }
+
+    #[serde_as]
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct Auction {
+        pub id: u64,
+        pub block: u64,
+        pub orders: Vec<Order>,
+        #[serde_as(as = "BTreeMap<_, DisplayFromStr>")]
+        pub prices: BTreeMap<H160, BigUint>,
+    }
+
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct Order {
+        // TODO: what fields? Needs to be documented in openapi too.
+    }
```

Comment:
This was left open in the PR that created the openapi docs. Here We can start using a smaller order model. Can add that in a separate PR.

Reply:
From what I can gather from `http_solver/model.rs` the data that's required for the external solvers is: uid, sell_token, buy_token, sell_amount, buy_amount, partially_fillable, kind, class, reward, executed_buy_amount, executed_sell_amount_before_fees, full_fee_amount, fee_amount, surplus_fee (part of class).
This should be the least amount of information the driver gets from the `protocol`.
Currently we also need `creation_date` but it makes sense to replace that with `is_mature` in the `protocol` (or simply not forward immature orders).

Reply:
Actually let me try commenting out individual fields of the current order until compiling the solver binary fails. That should give us a good idea.

Reply:
To the driver, we need enough data in order to build the transaction. This means that the full `OrderData` struct (the order parameters that are signed) as well as `Signature` are needed.

Order UID is derivable by order data + signature, but it might make sense to include as a convenience.

In terms of metadata, the fields that affect the driver are:
- `full_fee_amount` for 
- `is_mature` (I would pass this instead of creation dates, so that the `autopilot` decides on maturity). Hopefully, we can get rid of this though...
- `class` also needed for objective value computation
- `surplus_fee` for effective limit price computation
- `executed_amount` for partially fillable orders.